diff --git a/util/grub.d/10_linux.in b/util/grub.d/10_linux.in
index b70dca275..75ba29f95 100644
--- a/util/grub.d/10_linux.in
+++ b/util/grub.d/10_linux.in
@@ -179,6 +179,10 @@ mktitle ()
       title=$(printf '%s (%s) %s (recovery mode)' \
                      "${OS_NAME}" "${version}" "${OS_VERS}")
       ;;
+    fallback)
+      title=$(printf '%s (%s) %s (fallback mode)' \
+                     "${OS_NAME}" "${version}" "${OS_VERS}")
+      ;;
     *)
       title=$(printf '%s (%s) %s' \
                      "${OS_NAME}" "${version}" "${OS_VERS}")
@@ -308,7 +312,7 @@ while [ "x$list" != "x" ] ; do
   basename=`basename $linux`
   dirname=`dirname $linux`
   rel_dirname=`make_system_path_relative_to_its_root $dirname`
-  version=`echo $basename | sed -e "s,^[^0-9]*-,,g"`
+  version=`echo $basename | sed -e "s,vmlinuz-,,g"`
   alt_version=`echo $version | sed -e "s,\.old$,,g"`
   linux_root_device_thisversion="${LINUX_ROOT_DEVICE}"
 
@@ -410,6 +414,17 @@ while [ "x$list" != "x" ] ; do
                 "${GRUB_CMDLINE_LINUX} ${GRUB_CMDLINE_LINUX_DEFAULT} ${GRUB_CMDLINE_LINUX_DEBUG}"
   fi
 
+  if test -e "${dirname}/initramfs-${version}-fallback.img" ; then
+    initrd="initramfs-${version}-fallback.img"
+
+    if test -n "${initrd}" ; then
+      gettext_printf "Found fallback initrd image(s) in %s:%s\n" "${dirname}" "${initrd_extra} ${initrd}" >&2
+    fi
+
+    linux_entry "${OS}" "${version}" fallback standard \
+                "${GRUB_CMDLINE_LINUX} ${GRUB_CMDLINE_LINUX_DEFAULT}"
+  fi
+
   if [ "x${GRUB_DISABLE_RECOVERY}" != "xtrue" ]; then
     linux_entry "${OS}" "${version}" recovery standard \
                 "single ${GRUB_CMDLINE_LINUX}"
